oap-server/server-storage-plugin/storage-elasticsearch-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/elasticsearch/query/MetadataQueryEsDAO.java

```diff
@@ -191,6 +195,37 @@ public ServiceInstance getInstance(final String instanceId) throws IOException {
         return endpoints;
     }
 
+    @Override
+    public List<Process> listProcesses(String serviceId, String instanceId) throws IOException {
+        final String index =
+                IndexController.LogicIndicesRegister.getPhysicalTableName(ProcessTraffic.INDEX_NAME);
+
+        final BoolQueryBuilder query = Query.bool();
+        final SearchBuilder search = Search.builder().query(query).size(queryMaxSize);
+        if (StringUtil.isNotEmpty(serviceId)) {
+            query.must(Query.term(ProcessTraffic.SERVICE_ID, serviceId));
+        }
+        if (StringUtil.isNotEmpty(instanceId)) {
+            query.must(Query.term(ProcessTraffic.INSTANCE_ID, instanceId));
+        }
```

Comment:
I think serviceId and instanceId are both required.

Reply:
In the [query comment](https://github.com/apache/skywalking-query-protocol/blob/master/metadata-v2.graphqls#L136-L137), I mentioned they only providing one field is enough. It's not necessary to provide two fields, also the instance id contains the appoint service id. why do we need to provide two?

Reply:
I see.